### PR TITLE
Remove Timex dependency

### PR DESCRIPTION
**Pull request by rodrigues**
This removes Timex dependency and bases Exq solely on Elixir 1.3 Calendar types 🙂

I've started this effort for having trouble trying to run Exq on Elixir 1.3.

[I have some errors when running tests locally](https://gist.github.com/rodrigues/5238df7c9d1f6ef97d0971f642f3ea13), though they look a little unrelated with Timex.

[Travis is green](https://travis-ci.org/akira/exq/jobs/148536258) on `Elixir 1.3.2, OTP 17.5+`. Previous versions fail, as expected.

### CODE REVIEW

**Comment by baron** [comment]
Having mucked around with the recent changes to Timex I think this would be a great change and future proof the exq lib. Timex is a great library but it's a little overkill considering the recent improvements to elixir's native implementation and Timex itself went through some drastic api changes in the most recent version.

File: lib/exq/redis/job_queue.ex [hunk 1]
```diff
@@ -223,24 +227,29 @@ defmodule Exq.Redis.JobQueue do
...
 
       # Similar to Sidekiq strategy
       offset = :math.pow(job.retry_count, 4) + 15 + (Randomize.random(30) * (job.retry_count + 1))
-      time = Time.add(Time.now, Time.from(offset * 1_000_000, :microseconds))
+      time = offset_from_now(offset)
       Logger.info("Queueing job #{job.jid} to retry in #{offset} seconds")
       enqueue_job_at(redis, namespace, Job.to_json(job), job.jid, time, retry_queue_key(namespace))
   end
 
+  defp offset_from_now(offset) do
```
Comment (akira) [inline lib/exq/redis/job_queue.ex]: Can we move these time handling functions to a separate utility?

File: lib/exq/redis/job_stat.ex [hunk 2]
```diff
@@ -117,11 +114,16 @@ defmodule Exq.Redis.JobStat do
   end
 
   defp format_current_date(current_date) do
```
Comment (akira) [inline lib/exq/redis/job_stat.ex]: Same, can we move this to utility

File: .travis.yml [hunk 3]
```diff
@@ -3,9 +3,11 @@ elixir:
   - 1.0.5
   - 1.1.1
   - 1.2.0
```
Comment (akira) [inline .travis.yml]: Need to remove elixir 1.2.0 and earlier, since DateTime is only in Elixir 1.3

**Comment by akira** [comment]
@rodrigues This looks good, I wanted to do some additional testing though before I merge it in. Also, will probably do a change to the https://github.com/akira/exq_ui repo as well to remove Timex (only 1 use of it there).   Thanks!!

**Comment by rodrigues** [comment]
@akira 👍😉

**Comment by akira** [comment]
@rodrigues tested and looks good, merged to master.  Thanks again! ✨✨

**Pull request merged**